### PR TITLE
Add the 2D Array data type

### PR DESCRIPTION
**Pull request by ADawesomeguy**
I added the `2da` filetype which is used in some of BioWare's game engines and in games such as the first Witcher.

## Description
Not really much more to describe. For some reason there where changes made to other areas which was either a Git mistake by me or an automated script thing.

## Checklist:
- [x] **I am adding a new language.**
	- [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
		- [Search results for `.2da` extension](https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3A2da+2DA+NOT+nothack)
	- [x] **I have included a real-world usage sample for all extensions added in this PR:**
		- **Source:** [`Jaysyn904/RATDOG`](https://github.com/Jaysyn904/RATDOG)
		- **License:** CC0-1.0
	- [x] **I have included a syntax highlighting grammar:**
		- **Previews:** [[ 1 ](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2Fmaster%2Fgrammars%2F2da.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2F714c753b34b24df73b8f0bb4063a2c85c0e2aadf%2Fsamples%2Fitmwizranged.2da&code=)[| 2 ](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2Fmaster%2Fgrammars%2F2da.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2FGibberlings3%2FSwordCoastStratagems%2Fblob%2F2efda3413873f8557c459b13f92a2fe1882d8782%2Fstratagems%2Flib%2Fvanilla_hlas.2da&code=)[| 3 ](https://github-lightshow.herokuapp.com/?utf8=%E2%9C%93&scope=from-url&grammar_format=auto&grammar_url=https%3A%2F%2Fgithub.com%2FAlhadis%2Flanguage-etc%2Fblob%2Fmaster%2Fgrammars%2F2da.cson&grammar_text=&code_source=from-url&code_url=https%3A%2F%2Fgithub.com%2Flayonara%2Fnwn-haks%2Fblob%2F3bba79eaadc591479f44275cb0de7ec97c096c54%2Fconfig%2F2da%2Fskyboxes.2da&code=)]
		- **Source:** [`Alhadis/language-etc@714c753`](https://github.com/Alhadis/language-etc/commit/714c753b34b24df73b8f0bb4063a2c85c0e2aadf)
	- [ ] ~~I have updated the heuristics to distinguish my language from others using the same extension.~~

### CODE REVIEW

**Comment by Nixinova** [comment]
Your clone is out of date.
Also the [samples/2D Array/appearance.2da](https://github.com/github/linguist/pull/5785/files#diff-1b0948f9d073cbb5913b0af5a30840532c00c6149117b7b66d3bb3cb844f57bb) file is way too big.

File: grammars.yml [hunk 1]
```diff
@@ -703,6 +703,11 @@ vendor/grammars/latex.tmbundle:
 vendor/grammars/linter-lilypond:
 - source.lilypond
 vendor/grammars/liquid-tm-grammar:
+- liquid.injection
+- source.css
+- source.js
+- source.json
+- text.html.basic
 - text.html.liquid
```
Comment (Alhadis) [inline grammars.yml:711]: Remove these added lines, please. The only entry in this list should be `text.html.liquid`:

```suggestion
- text.html.liquid
```

File: grammars.yml [hunk 2]
```diff
@@ -966,6 +971,15 @@ vendor/grammars/vscode-hack:
...
+- source.bnd
+- source.cl
+- source.dds.dspf
+- source.dds.icff
+- source.dds.lf
+- source.dds.pf
+- source.dds.prtf
+- source.mi
+- source.rpg
 - source.rpgle
```
Comment (Alhadis) [inline grammars.yml:983]: See above. This list should only contain `source.rpgle`:

```suggestion
- source.rpgle
```

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -44,6 +44,14 @@
   tm_scope: source.bsl
   ace_mode: text
   language_id: 0
+2D Array:
```
Comment (Alhadis) [inline lib/linguist/languages.yml:47]: This should actually be named `2-Dimensional Array`. BioWare's documentation consistently refers to this format as either _"2da"_ or _"2-dimensional array"_, but rarely _"2D Array"_.

File: vendor/README.md [hunk 4]
```diff
@@ -7,6 +7,7 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
```
Comment (Alhadis) [inline vendor/README.md:7]: All the changes you've made to this file should be regenerated by running `script/list-grammars`. Make sure to do so after renaming the language entry.

**Review by Alhadis — CHANGES_REQUESTED** [review]
It's actually [BioWare's format](https://nwn.fandom.com/wiki/.2da), not Bethesda's (unless they've been using the Infinity Engine when I wasn't looking…).

Also, as pointed out by @Nixinova, you'll need to provide (much) smaller samples. Truncating each of the files you've provided to the first 100 lines should be more than enough.

**Comment by ADawesomeguy** [comment]
Alright cool, I accidentally mistyped lol.

**Comment by ADawesomeguy** [comment]
I'll fix all the changes you requested and push tomorrow because it's kinda late for me tonight.

File: lib/linguist/languages.yml (hunk 3 above)
Comment (ADawesomeguy) [reply lib/linguist/languages.yml]: Changed it an reran the list grammars script.

File: grammars.yml (hunk 1 above)
Comment (ADawesomeguy) [reply grammars.yml]: I removed these manually as the extra commit will be squashed anyways.

**Comment by ADawesomeguy** [comment]
Alright, I think everything's fixed, please let me know if there's anything else you need me to do.

**Comment by ADawesomeguy** [comment]
@Alhadis I removed the conflicting scope, feel free to modify this PR to include your grammar and rearrange the submodules :)

**Review by Alhadis — APPROVED** [review]

**Comment by Alhadis** [comment]
Just did. 😉 Thanks for your patience!

**Review by lildude — APPROVED** [review]

**Pull request merged**